src/cs/building/PowerPlant.cs

```diff
@@ -332,4 +342,16 @@ public partial class PowerPlant : Node2D {
 		// Update the UI
 		_UpdatePlantData();
 	}
+	
+	private void _on_area_2d_mouse_entered() {
+		if(!IsPreview) {
+			NameR.Show();
+		}
+		
+	}
+	private void _on_area_2d_mouse_exited() {
+		if(!IsPreview) {
+			NameR.Hide();
+		}
```

Comment:
It looks like you connected these signals in the editor. To make all objects truly generic, I've opted for connecting all signals in the code for all objects. It will be good to have that here too. Also please follow c# naming conventions (PascalCase) and not the gdscript (snake_case) ones.